dask/dataframe/io/parquet/arrow.py

```diff
@@ -168,6 +168,12 @@ def read_metadata(
                 storage_name_mapping,
                 column_index_names,
             ) = _parse_pandas_metadata(pandas_metadata)
+            categories = categories or []
+            for col in pandas_metadata["columns"]:
+                if (col["pandas_type"] == "categorical") and (
+                    col["name"] not in categories
+                ):
+                    categories.append(col["name"])
```

Comment:
In `read_parquet`, we say that `categories` can also be a dict mapp

```
    categories : list, dict or None
        For any fields listed here, if the parquet encoding is Dictionary,
        the column will be created with dtype category. Use only if it is
        guaranteed that the column is encoded as dictionary in all row-groups.
        If a list, assumes up to 2**16-1 labels; if a dict, specify the number
        of labels expected; if None, will load categories automatically for
        data written by dask/fastparquet, not otherwise.
```

Perhaps by the time it gets here, we've ensured that it's a list or None, but can you verify?

I'm not sure that the pyarrow reader should necessarily implement the same API FWIW.

Reply:
Good point!  I think it is the last line of the description that we should align with here:
> if None, will load categories automatically for data written by dask/fastparquet, not otherwise.

I interpret this to mean that we should only use the pandas metadata to automatically populate `categories` (as a list), if it was originally set to `None`